scripts/icu/63.1-min-static-data/script.sh

```diff
@@ -0,0 +1,160 @@
+
+        pushd ${MASON_ROOT}/..
+        env -i HOME="$HOME" PATH="$PATH" USER="$USER" ${MASON_DIR}/mason build ${MASON_NAME} ${MASON_VERSION}
+        popd
+
+        # TODO: Copies a bunch of files to a kind of orphaned place, do we need to do something to clean up after the build?
+        #  Copying the whole build directory is the easiest way to do a cross build, but we could limit this to a small subset of files (icucross.mk, the tools directory, probably a few others...)
+        #  Also instead of using the regular build steps, we could use a dedicated built target that just builds the tools
+        mason_substep "Moving host ICU build directory to ${MASON_ROOT}/.build/icu-host"
+        rm -rf ${MASON_ROOT}/.build/icu-host
+        cp -R ${MASON_BUILD_PATH}/source ${MASON_ROOT}/.build/icu-host
+    fi
+}
+
+function trim_data {
+    mason_substep "Trimming ICU data to small number of locals and datasets"
+
+    # for some reason there is also a dat file in there!
+    rm -f data/in/icudt63l.dat
+
+    # move all non algorithmic code point mappings so they arent used
+    for f in $(find data/mappings -name '*.mk'); do
+        mv $f ${f}_
+    done
+
+    # make local versions of each data set
+    for f in $(find data -path mappings -prune -o -name '*files.mk'); do
+        # make a local version and remove the line continuations
+        l=$(echo $f | sed -e "s/\(.*\)files.mk/\1local.mk/g")
+        sed -e :a -e '/\\$/N; s/\\\n//; ta' $f > $l
+
+        # if its locale, unit or currency we use our supported lang list
+        if [ "$(echo $l | grep -cE 'locales|unit|curr')" -eq 1 ]; then
+            sed -i'' -e '/^#/!s/SOURCE.*=.*txt/SOURCE = da.txt de.txt en.txt eo.txt es.txt fi.txt fr.txt he.txt id.txt it.txt ko.txt my.txt nl.txt pl.txt pt.txt pt_PT.txt ro.txt ru.txt sv.txt tr.txt uk.txt vi.txt zh.txt zh_Hans.txt/g' $l
+        # if its misc we need a couple of things
+        elif [ $(echo $l | grep -cF "misc") -eq 1 ]; then
+            sed -i'' -e '/^#/!s/SOURCE.*=.*txt/SOURCE = numberingSystems.txt icuver.txt icustd.txt/g' $l
+        # otherwise the scuttle the whole thing
+        else
+            sed -i'' -e '/^#/!s/SOURCE.*=.*txt/SOURCE = /g' $l
+        fi
+    done
+}
+
+function mason_compile {
+    if [[ ${MASON_PLATFORM} == 'ios' || ${MASON_PLATFORM} == 'android' || ${MASON_PLATFORM_VERSION} != `uname -m` ]]; then
+        MASON_CROSS_BUILD=1
+    fi
+    mason_compile_base
+}
+
+function mason_compile_base {
+    pushd  ${MASON_BUILD_PATH}/source
+    
+    # trim out a bunch of the data so that the data static library is as small as possible
+    trim_data
+
+    # Using uint_least16_t instead of char16_t because Android Clang doesn't recognize char16_t
+    # I'm being shady and telling users of the library to use char16_t, so there's an implicit raw cast
+    ICU_CORE_CPP_FLAGS="-DU_CHARSET_IS_UTF8=1 -DU_CHAR_TYPE=uint_least16_t"
```

Comment:
@kevinkreiser this `U_CHAR_TYPE` customization (and the below `-DUCHAR_TYPE=char16_t` in the `mason_cflags`) was done by @ChrisLoer for MBGL and may not benefit your usecase, so I would consider removing it unless you specific think it is helpful for your usecase.

Reply:
FWIW, mbgl-core is no longer using mason for ICU, so totally fine for the mason build to stop using the `U_CHAR_TYPE` customization.

Reply:
indeed, we looked at vendoring as well (which i think we want to do long term) but decided against biting the bullet of figuring out how to vendor the data into the lib (so we can ship without having to bundle the data separately)